pennylane_qiskit/qiskit_device2.py

```diff
@@ -495,7 +505,10 @@ def execute_circuits(session):
                             f"Setting shot vector {circ.shots.shot_vector} is not supported for {self.name}."
                             f"The circuit will be run once with {circ.shots.total_shots} shots instead."
                         )
-                    if isinstance(circ.measurements[0], (ExpectationMP, VarianceMP)):
+                    if (
+                        isinstance(circ.measurements[0], (ExpectationMP, VarianceMP))
+                        and circ.measurements[0].obs.pauli_rep
```

Comment:
```suggestion
                        and getattr(circ.measurements[0], "pauli_rep", None)
```
I prefer `getattr` in such scenarios so that we don't get attribute not present errors in some edge cases.